### DISCUSSION TITLE
Custom text on edges and faces (Viewer Index)

### DISCUSSION THREAD

**Discussion by Durman** [nortikin/sverchok#4566] [Ideas]
Currently, when text socket of the Viewer Index node is connected the node, instead of indexes, displays data from the socket on vertices of input mesh. It seems logically to propagate the same behavior to edges and faces. Now they show indexes even when the text socket is connected.

![image](https://user-images.githubusercontent.com/28003269/178660082-e4fa517c-8bcd-4fbe-8a76-d81db2a40f1b.png)

**Comment by zeffii** [comment]
the Text socket could have a feature that lets users switch between verts/edges/faces for the location of the received Text,  you mean something like that?

**Reply by Durman** [reply]
I don't think such option would have too much sense. It seems quite unlike use case for the node to show custom text on one type of elements and indexes on two other types.
The proposal is to show either indexes or custom text on all type of elements.

**Reply by zeffii** [reply]
> The proposal is to show either indexes or custom text on all type of elements.

right, no issue there, the question should have been :   how do you propose to change the interface to facilitate this?

**Reply by Durman** [reply]
No changes are required. If text socket is connected the custom text is shown on vertices, edges and faces. If it is not connected then indexes are shown.

**Comment by Durman** [comment]
Done #4595
